pkg/apitype/errors.go

```diff
@@ -16,11 +16,56 @@ package apitype
 
 import "fmt"
 
+// ErrorType is an enum for various types of common errors that occur.
+type ErrorType string
+
+const (
+	// NotFoundErrorType is used when a resource or resource field was not found.
+	// For example, updating a Stack that doesn't exist.
+	NotFoundErrorType ErrorType = "not_found"
+	// RequiredErrorType is used when a resource or resource field is missing and is required.
+	// For example, creating a Stack without a project name.
+	RequiredErrorType ErrorType = "required"
+	// InvalidErrorType is used when the a resource or field was passed with an invalid value or state.
+	// For example, creating a Stack with an invalid name.
+	InvalidErrorType ErrorType = "invalid"
+	// AlreadyExistsErrorType is used if the resource or field already exists (and must be unique).
+	// For example, creating two Stacks with the same name.
+	AlreadyExistsErrorType ErrorType = "already_exists"
+
+	// CustomErrorType is used to describe any ErrorType not found in this file, and must be paired with
+	// a custom error message.
+	CustomErrorType ErrorType = "custom"
+)
+
+// RequestError describes a request error in more detail, such the specific validation
+// error(s) that caused the request to fail and links to the relevant documentation.
+type RequestError struct {
+	// Resource is the user-friendly description of the resource, e.g. "stack" or "tag".
+	Resource string `json:"resource"`
+
+	// Attribute describes the property of the resource (if applicable) that is problematic, e.g.
+	// "name" or "length".
+	Attribute *string `json:"attribute,omitempty"`
```

Comment:
Would it make sense for this to be a list of strings, if we have multiple attributes that are invalid?

Reply:
Oh, I see - there can be multiple `RequestErrors` on a single error response.

Reply:
Yes, I think we would just add a separate `RequestError` for every field that was invalid in some way.